Tests/Common/Data/VolumeRenkoConsolidatorTests.cs

```diff
@@ -0,0 +1,258 @@
+                bar = consolidated;
+            };
+
+            var reference = new DateTime(2013, 10, 1);
+            consolidator.Update(new Tick(reference, Symbol.Empty, String.Empty, String.Empty, 2m, 1m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(1), Symbol.Empty, String.Empty, String.Empty, 3m, 2m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(2), Symbol.Empty, String.Empty, String.Empty, 3m, 3m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(3), Symbol.Empty, String.Empty, String.Empty, 2m, 2m));
+            Assert.IsNotNull(bar);
+
+            Assert.AreEqual(1m, bar.Open);
+            Assert.AreEqual(3m, bar.High);
+            Assert.AreEqual(1m, bar.Low);
+            Assert.AreEqual(2m, bar.Close);
+            Assert.AreEqual(10m, bar.Volume);
+            Assert.AreEqual(10m, bar.BrickSize);
+            Assert.AreEqual(Symbol.Empty, bar.Symbol);
+            Assert.AreEqual(reference, bar.Start);
+            Assert.AreEqual(reference.AddHours(3), bar.EndTime);
+            Assert.IsTrue(bar.IsClosed);
+
+            consolidator.Update(new Tick(reference.AddHours(4), Symbol.Empty, String.Empty, String.Empty, 4m, 1m));
+
+            consolidator.Update(new Tick(reference.AddHours(5), Symbol.Empty, String.Empty, String.Empty, 3m, 2m));
+
+            consolidator.Update(new Tick(reference.AddHours(6), Symbol.Empty, String.Empty, String.Empty, 4m, 3m));
+
+            Assert.AreEqual(2m, bar.Open);
+            Assert.AreEqual(3m, bar.High);
+            Assert.AreEqual(1m, bar.Low);
+            Assert.AreEqual(3m, bar.Close);
+            Assert.AreEqual(10m, bar.Volume);
+            Assert.AreEqual(10m, bar.BrickSize);
+            Assert.AreEqual(Symbol.Empty, bar.Symbol);
+            Assert.AreEqual(reference.AddHours(3), bar.Start);
+            Assert.AreEqual(reference.AddHours(6), bar.EndTime);
+            Assert.IsTrue(bar.IsClosed);    // bar is always closed since it is the consolidated bar instance
+
+            consolidator.Update(new Tick(reference.AddHours(7), Symbol.Empty, String.Empty, String.Empty, 5m, 10m));
+
+            // Not yet consolidated, so bar is not updated yet
+            Assert.AreEqual(2m, bar.Open);
+            Assert.AreEqual(3m, bar.High);
+            Assert.AreEqual(1m, bar.Low);
+            Assert.AreEqual(3m, bar.Close);
+            Assert.AreEqual(10m, bar.Volume);
+            Assert.AreEqual(10m, bar.BrickSize);
+            Assert.AreEqual(Symbol.Empty, bar.Symbol);
+            Assert.AreEqual(reference.AddHours(3), bar.Start);
+            Assert.AreEqual(reference.AddHours(6), bar.EndTime);
+            Assert.IsTrue(bar.IsClosed);
+        }
+
+        [TestCase(Language.CSharp)]
+        [TestCase(Language.Python)]
+        public void SelectorCanBeOptionalWhenVolumeSelectorIsPassed(Language language)
```

Comment:
Don't think this test makes sense, there's no selector for `VolumeRenkoConsolidator` vs `ClassicRenkoConsolidator` which does have. The following question is would it make sense for `VolumeRenkoConsolidator` to have a volume selector, why does the classic consolidator have one?

Reply:
Yah it don't really make sense for it. The classic one does because volume is stacked in a different way then the other attributes. I'll remove it.